pyomo/contrib/parmest/parmest.py

```diff
@@ -662,9 +719,12 @@ def theta_est(self, solver="ef_ipopt", return_values=[], calc_cov=False, cov_n=N
         calc_cov: boolean, optional
             If True, calculate and return the covariance matrix (only for "ef_ipopt" solver)
         cov_n: int, optional
-            If calc_cov=True, then the user needs to supply the number of datapoints 
+            If calc_cov=True, then the user needs to supply the number of datapoints
             that are used in the objective function
-            
+        pre_solve: boolean, optional
+            If True, solve square problem with parameters fixed before solving parameter estimation
+        bound_push: float, optional
+            If True, set ipopt solver option 'bound_push' to value provided
```

Comment:
The `bound_push` option is too specific to IPOPT. I recommend generalizing this to something like `solver_options` instead of assuming the solver will always be IPOPT

Reply:
@blnicho yes, I appreciate that `bound_push` is IPOPT-specific so, to avoid confusion, I have updated the name of the optional argument to `initialization_with_bound_push`.  I am adding this as a separate argument to `solver_options` since IPOPT is hard-coded as the solver in the method `_Q_opt()` in parmest.py and in the method `inv_reduced_hessian_barrier()` in inverse_reduced_hessian.py